src/libaktualizr/utilities/utils.h

```diff
@@ -44,10 +43,19 @@ struct Utils {
   static int shell(const std::string &command, std::string *output, bool include_stderr = false);
   static boost::filesystem::path absolutePath(const boost::filesystem::path &root, const boost::filesystem::path &file);
   static void setSocketPort(sockaddr_storage *addr, in_port_t port);
-  static std::vector<boost::filesystem::path> glob(const std::string &pat);
   static void createDirectories(const boost::filesystem::path &path, mode_t mode);
   static std::string urlEncode(const std::string &input);
   static CURL *curlDupHandleWrapper(CURL *curl_in, bool using_pkcs11);
+  static std::vector<boost::filesystem::path> getDirEntriesByExt(const boost::filesystem::path &dir_path,
+                                                                 const std::string &ext);
+#if defined(ANDROID)
+  static void setStorageRootPath(const std::string &storage_root_path);
+  static boost::filesystem::path getStorageRootPath();
+  static std::mutex storage_root_path_mutex_;
```

Comment:
Does it really need a mutex? I would have thought that `setStorageRootPath()` would be called once in early initialization.

I also think we could forego the `#if defined` here and just make this api available in all cases, with a suitable comment (like "used for Android support").
The code in `SafeTempRoot` could default to use this path if not empty and attempt to use a temporary directory otherwise.

So, `if (!Utils::getStorageRootPath())` instead of `#if defined(ANDROID)`.

Reply:
Yes, that's a better idea.